src/diagnostics/Diagnostics.jl

```diff
@@ -0,0 +1,40 @@
+module Diagnostics
+
+import LinearAlgebra: dot
+
+import ClimaCore: Fields, Geometry, InputOutput, Meshes, Spaces, Operators
+import ClimaCore.Utilities: half
+import Thermodynamics as TD
+
+import ..AtmosModel
+import ..call_every_n_steps
+import ..Parameters as CAP
+
+import ..unit_basis_vector_data
+
+# moisture_model
+import ..DryModel
+import ..EquilMoistModel
+import ..NonEquilMoistModel
+
+# energy_form
+import ..TotalEnergy
+
+# turbconv_model
+import ..EDMFX
+import ..DiagnosticEDMFX
+
+# Abbreviations (following utils/abbreviations.jl)
+const curlₕ = Operators.Curl()
+const CT3 = Geometry.Contravariant3Vector
+const ᶜinterp = Operators.InterpolateF2C()
+# TODO: Implement proper extrapolation instead of simply reusing the first
+# interior value at the surface.
+const ᶠinterp = Operators.InterpolateC2F(
+    bottom = Operators.Extrapolate(),
+    top = Operators.Extrapolate(),
+)
```

Comment:
Can we just reuse `utils/abbreviations.jl`? (and do `import ..`)

Reply:
If we want to do so, we should move out the definitions of `AxisTensor` and `convert` from `abbreviations.jl`, otherwise we get a redefinition.

Reply:
Ah, right, we need to get rid of those..